[Metrics UI] Add back context variables with descriptions

phillipb opened the pull request:
## Summary

Add back the old context variables with clearer names and descriptions. Fixes: #65974.

<img width="227" alt="Screen Shot 2020-05-27 at 9 33 09 AM" src="https://user-images.githubusercontent.com/1482089/83033751-1e2c8d80-9ffd-11ea-91ac-53cb7b1384f8.png">


### Checklist

Delete any items that are not applicable to this PR.

- [x] Any text added follows [EUI's writing guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses sentence case text and includes [i18n support](https://github.com/elastic/kibana/blob/master/packages/kbn-i18n/README.md)

elasticmachine commented:
Pinging @elastic/logs-metrics-ui (Team:logs-metrics-ui)

phillipb commented:
@elasticmachine merge upstream

sorantis commented:
@phillipb From the user's perspective it's not clear that each of these context variables is followed by `conditionN`. Some users reported the following output, which indicated that they didn't use these variables correctly.

```
alertid:c6b3c440-5d14-49be-bc16-9dae92386ffe
alertName:test
spaceId:default
tags:test
alertInstanceId:304ef5a3-a6b9-4ee3-8ffb-8e2742bd3a7b-hayate
context.group:test
context.valueOf:[object Object]
context.thresholdOf:[object Object]
context.metricOf:[object Object]
```
Can we clarify it somehow? Maybe by default add condition0 to each context variable as soon as the user clicks on the dropdown? Thoughts?

phillipb commented:
@sorantis yeah, I added this issue to address that https://github.com/elastic/kibana/issues/67396. It's now added to 7.9 as a documentation fix. 

Because the number of conditions is dynamic and the context variable list is static, not sure how we could create a better UX. We could hardcode some values in the context for example `ctx.value.condition0`, `ctx.value.condition1`, but that would fail if there were only one condition.

kibanamachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/50047/)
* Commit: 51e2f7480f1fb548cb5b21d5f113acd772b666cc
* [Pipeline Steps](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/50047/flowGraphTable) (look for red circles / failed steps)
* [Interpreting CI Failures](https://www.elastic.co/guide/en/kibana/current/interpreting-ci-failures.html)

### Failed CI Steps
* [Execute kibana-intake](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/50047/execution/node/251/log)

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/50047/","number":"50047","commit":"51e2f7480f1fb548cb5b21d5f113acd772b666cc"}]}
PIPELINE-->

sorantis commented:
By all means we should update our docs. I'm wondering if there's any way we could put this information on the UI too. It feels like the docs will be helpful only after the user has already struggled with making it work :)

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/50078/)
* Commit: 3752dc319ea5fba1005126bb76a7e2105af6657a

### History
* :broken_heart: [Build #50047](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/50047/) failed 51e2f7480f1fb548cb5b21d5f113acd772b666cc

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/50047/","number":"50047","commit":"51e2f7480f1fb548cb5b21d5f113acd772b666cc"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/50078/","number":"50078","commit":"3752dc319ea5fba1005126bb76a7e2105af6657a"}]}
PIPELINE-->

phillipb commented:
@sorantis we could link to think docs from the flyout maybe? We could also file an issue with kibana-alerting to see if they have thoughts or plans around this.

phillipb commented:
@elasticmachine merge upstream

kibanamachine commented:
## :yellow_heart: Build succeeded, but was flaky
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/50784/)
* Commit: 3974beeb5d92f5b7db3a8387e385f76928b36369
* Flaky suites:
  * xpack-kibana-ciGroup10

### History
* :green_heart: [Build #50078](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/50078/) succeeded 3752dc319ea5fba1005126bb76a7e2105af6657a
* :broken_heart: [Build #50047](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/50047/) failed 51e2f7480f1fb548cb5b21d5f113acd772b666cc

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/50047/","number":"50047","commit":"51e2f7480f1fb548cb5b21d5f113acd772b666cc"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/50078/","number":"50078","commit":"3752dc319ea5fba1005126bb76a7e2105af6657a"},{"status":"UNSTABLE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/50784/","number":"50784","commit":"3974beeb5d92f5b7db3a8387e385f76928b36369"}]}
PIPELINE-->

Zacqary commented on x-pack/plugins/infra/server/lib/alerting/metric_threshold/register_metric_threshold_alert_type.ts:
```diff
@@ -55,6 +55,29 @@ export function registerMetricThresholdAlertType(libs: InfraBackendLibs) {
+    'xpack.infra.metrics.alerting.threshold.alerting.valueActionVariableDescription',
+    {
+      defaultMessage:
+        'The value of the metric in the specified condition. Usage: (ctx.value.condition0).',
```
Maybe `Usage: (ctx.value.condition0, ctx.value.condition1, etc...)`?

phillipb replied:
works for me.

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/50858/)
* Commit: ca0b80043541388751d28317d5ae501340dcb2ec

### History
* :yellow_heart: [Build #50784](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/50784/) was flaky 3974beeb5d92f5b7db3a8387e385f76928b36369
* :green_heart: [Build #50078](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/50078/) succeeded 3752dc319ea5fba1005126bb76a7e2105af6657a
* :broken_heart: [Build #50047](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/50047/) failed 51e2f7480f1fb548cb5b21d5f113acd772b666cc

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/50047/","number":"50047","commit":"51e2f7480f1fb548cb5b21d5f113acd772b666cc"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/50078/","number":"50078","commit":"3752dc319ea5fba1005126bb76a7e2105af6657a"},{"status":"UNSTABLE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/50784/","number":"50784","commit":"3974beeb5d92f5b7db3a8387e385f76928b36369"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/50858/","number":"50858","commit":"ca0b80043541388751d28317d5ae501340dcb2ec"}]}
PIPELINE-->

phillipb commented:
@elasticmachine merge upstream

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/51727/)
* Commit: bfe80921e11c509858e13ae89e52cf64550cc787

### History
* :green_heart: [Build #50858](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/50858/) succeeded ca0b80043541388751d28317d5ae501340dcb2ec
* :yellow_heart: [Build #50784](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/50784/) was flaky 3974beeb5d92f5b7db3a8387e385f76928b36369
* :green_heart: [Build #50078](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/50078/) succeeded 3752dc319ea5fba1005126bb76a7e2105af6657a
* :broken_heart: [Build #50047](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/50047/) failed 51e2f7480f1fb548cb5b21d5f113acd772b666cc

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/50047/","number":"50047","commit":"51e2f7480f1fb548cb5b21d5f113acd772b666cc"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/50078/","number":"50078","commit":"3752dc319ea5fba1005126bb76a7e2105af6657a"},{"status":"UNSTABLE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/50784/","number":"50784","commit":"3974beeb5d92f5b7db3a8387e385f76928b36369"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/50858/","number":"50858","commit":"ca0b80043541388751d28317d5ae501340dcb2ec"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/51727/","number":"51727","commit":"bfe80921e11c509858e13ae89e52cf64550cc787"}]}
PIPELINE-->

kibanamachine commented:
Friendly reminder: Looks like this PR hasn’t been backported yet.
To create backports run `node scripts/backport --pr 67487` or prevent reminders by adding the `backport:skip` label.

kibanamachine commented:
Friendly reminder: Looks like this PR hasn’t been backported yet.
To create backports run `node scripts/backport --pr 67487` or prevent reminders by adding the `backport:skip` label.

kibanamachine commented:
Friendly reminder: Looks like this PR hasn’t been backported yet.
To create backports run `node scripts/backport --pr 67487` or prevent reminders by adding the `backport:skip` label.

kibanamachine commented:
Friendly reminder: Looks like this PR hasn’t been backported yet.
To create backports run `node scripts/backport --pr 67487` or prevent reminders by adding the `backport:skip` label.

kibanamachine commented:
Friendly reminder: Looks like this PR hasn’t been backported yet.
To create backports run `node scripts/backport --pr 67487` or prevent reminders by adding the `backport:skip` label.
